sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/checks/CheckTest.kt

```diff
@@ -0,0 +1,54 @@
+package org.sonarsource.kotlin.checks
+
+import org.junit.jupiter.api.Assumptions.assumeTrue
+import org.junit.jupiter.api.Test
+import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.verifier.KotlinVerifier
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.io.path.isRegularFile
+import kotlin.streams.asSequence
+
+private val checkNamesWithAvailableSampleTestFiles by lazy {
```

Comment:
by default `lazy` does synchronization and it might affect the performance. So it's better to use `lazy(LazyThreadSafetyMode.NONE)`